src/common/contexts/tokenInstancesContext.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { ReactElement } from "react";
+import { TitleEscrow } from "@govtechsg/token-registry/types/TitleEscrow";
+import { TradeTrustERC721 } from "@govtechsg/token-registry/types/TradeTrustERC721";
+import { WrappedDocument } from "@govtechsg/open-attestation";
+import { useTitleEscrowContract } from "../hooks/useTitleEscrowContract";
+import { useTokenRegistryContract } from "../hooks/useTokenRegistryContract";
+import { getDocumentId, getTokenRegistryAddress } from "../utils/document";
+import { useDefaultProvider } from "../hooks/useDefaultProvider";
+
+interface TokenInstanceContextType {
+  titleEscrowInstance?: TitleEscrow;
+  tokenRegistryInstance?: TradeTrustERC721;
+}
+
+export const TokenInstanceContext = React.createContext<TokenInstanceContextType>({
+  titleEscrowInstance: undefined,
+  tokenRegistryInstance: undefined,
+});
+
+export const TokenInstanceProvider = ({
+  document,
```

Comment:
maybe no need the word Instance at all? was confused about what instance meant.

TokenInstanceProvider change to TokenProvider, the same goes with all the other variables names, such as titleEscrowInstance, change to titleEscrow?